torchtitan/profiling.py

```diff
@@ -15,6 +16,14 @@
 # the number of warmup steps before the active step in each profiling cycle
 WARMUP = 3
 
+# how much memory allocation/free ops to record in memory snapshots
+MEMORY_SNAPSHOT_MAX_ENTRIES = 100000
```

Comment:
`MEMORY_SNAPSHOT_MAX_ENTRIES` controls how large `.pickle` can be. Right now `<50MB`. I will tune it by running 70B on mast as a follow up